### PR TITLE
Fetch pull request model for reviews in open state

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Verify Review status to fetch only pull request models for open reviews to avoid having duplicate links on GitHub comment.

Bot finds multiple entries in pull request model table but only one of them is linked to open reviews and rest all are linked to closed reviews.

Unfortunately cosmos DB doesn't allow join across two containers.

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Repositories/CosmosPullRequestsRepository.cs [hunk 1]
```diff
@@ -52,7 +54,22 @@ private async Task<List<PullRequestModel>> GetPullRequestFromQueryAsync(string q
                 var result = await itemQueryIterator.ReadNextAsync();
                 allRequests.AddRange(result.Resource);
             }
-            return allRequests;
+
+            // Cosmos doesn't allow cross join of two containers so we need to filter closed API reviews
```
Comment (weshaggard) [inline src/dotnet/APIView/APIViewWeb/Repositories/CosmosPullRequestsRepository.cs:58]: Are we worried about this becoming slow over time? If so perhaps we could create a view or projection of some sort on the db side.

**Review by weshaggard — APPROVED** [review]
looks reasonable.

**Comment by praveenkuttappan** [comment]
Fixes #3862

**Comment by praveenkuttappan** [comment]
Fixes #4035

File: src/dotnet/APIView/APIViewWeb/Repositories/CosmosPullRequestsRepository.cs (hunk 1 above)
Comment (praveenkuttappan) [reply src/dotnet/APIView/APIViewWeb/Repositories/CosmosPullRequestsRepository.cs:58]: Long term solution is to add a status field in pull request model entry as well or a trigger in the cosmos side.

**Pull request merged**